File: CoreAngularDemo.Client/src/app/modules/shared/models/document.ts
import { IssueLog } from './issuelog';
import { TEntity } from '../../core/models/entity/entity';
import { User } from './user';

export class Document extends TEntity<Document> {
  name?: string;
  description?: string;
  issueLog?: IssueLog;
  file?: File;
  mod?: User;
  updatedDate?: Date;
  newDate?: Date;
  path?: string;

  constructor(document: Partial<Document>) {
    super(document);
    this.issueLog = this.issueLog && new IssueLog(this.issueLog);
    this.mod = new User(this.mod);
  }
}

File: CoreAngularDemo.API/CoreAngularDemo.DAL/Models/Entities/Document.cs
using System;
using System.Collections.Generic;
using CoreAngularDemo.DAL.Models.Entities.Abstractions;

namespace CoreAngularDemo.DAL.Models.Entities
{
    public class Document : IAuditableEntity, IBaseEntity
    {
        public Document()
        {
            Bill = new HashSet<Bill>();
        }

        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int? IssueLogId { get; set; }
        public string ContentType { get; set; }
        public DateTime? CreatedDate { get; set; }
        public DateTime? UpdatedDate { get; set; }
        public string CreatedById { get; set; }
        public string UpdatedById { get; set; }
        public DateTime? NewDate { get; set; }
        public string Path { get; set; }
        public virtual User Create { get; set; }
        public virtual IssueLog IssueLog { get; set; }
        public virtual User Mod { get; set; }
        public virtual ICollection<Bill> Bill { get; set; }
    }
}
